packages/calling-component-bindings/src/hooks/useHandlers.ts

```diff
@@ -17,12 +17,25 @@ import { CallClientContext, useCall, useCallAgent, useDeviceManager } from '../p
 // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
 export const useHandlers = <PropsT>(component: (props: PropsT) => ReactElement | null) => {
   const callClient: StatefulCallClient = (useContext(CallClientContext) as any)?.callClient;
-  const callAgent = useCallAgent();
   const deviceManager = useDeviceManager();
-  const call = useCall();
+  const call = useContext(CallContext)?.call;
+  const callAgent = useContext(CallAgentContext)?.callAgent;
   if (!callClient) {
     return undefined;
   }
 
+  // Handle edge case, validate if call and callAgent are the same type (ACS/Teams)
+  if (callAgent && !isACSCallAgent(callAgent)) {
+    if (call && isACSCall(call)) {
+      throw new Error('A TeamsCall must be provided when callAgent is TeamsCallAgent');
+    }
+  }
+
+  if (callAgent && isACSCallAgent(callAgent)) {
+    if (call && !isACSCall(call)) {
```

Comment:
should we assume if it is not an acs call then its a teams call? I just have a light concern if we ever want to bring other types of calls (highly unlikely) but is it possible to do some other check instead of saying its just not an acs call?

Reply:
Agreed. I left similar comments elsewhere. A more future proof test would be:

```TypeScript

if (callAgent && isTeamsCallAgent(callAgent)) {
  if (call && !isTeamsCall(call) {
    throw new Error('Incompatible call type: A TeamsCall must be used with a TeamsCallAgent');
  }
}

if (callAgent && isAcsCallAgent(callAgent)) {
  if (call && !isAcsCall(call) {
    throw new Error('Incompatible call type: A Call must be used with a CallAgent');
  }
}
```

Reply:
Yep, that's a great point, in previous implementation, I didn't have a perfect way to implement isTeamsCall. But this isn't a problem anymore, lemme apply this comment in next a few iterations in the PR